lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -1361,7 +1361,8 @@ static SPIR::RefParamType transTypeDesc(Type *Ty,
     auto *ET = TPT->getElementType();
     SPIR::ParamType *EPT = nullptr;
     if (isa<FunctionType>(ET)) {
-      assert(isVoidFuncTy(cast<FunctionType>(ET)) && "Not supported");
+      // TODO: implement support for non void function type
+      // assert(isVoidFuncTy(cast<FunctionType>(ET)) && "Not supported");
```

Comment:
We have discussed it offline, but I still don't understand why we need to comment this assert and what are the consequences. While the PR is approved, I'd suggest to hold merge a bit until every one understands, what is happening here :)